go/tools/builders/cgo2.go

```diff
@@ -95,7 +98,27 @@ func cgo2(goenv *env, goSrcs, cgoSrcs, cSrcs, cxxSrcs, objcSrcs, objcxxSrcs, sSr
 		}
 	}
 	combinedLdFlags = append(combinedLdFlags, defaultLdFlags()...)
-	os.Setenv("CGO_LDFLAGS", strings.Join(combinedLdFlags, " "))
+
+	var ldflagsFile *os.File
+	combinedLdFlagsStr := strings.Join(combinedLdFlags, " ")
+	if len(combinedLdFlagsStr) < maxEnvVarSize {
```

Comment:
Let's also check here whether this is Go 1.23 or higher. There are a few calls to `runtime.Version()` in this directory that check. The `onVersion` function in `link.go` is probably the most general.